dsls/sql/src/main/java/org/apache/beam/dsls/sql/schema/BeamSqlUdaf.java

```diff
@@ -61,10 +65,17 @@ public BeamSqlUdaf(){}
 
   /**
    * get the coder for AccumT which stores the intermediate result.
-   * By default it's fetched from {@link CoderRegistry}.
+   * By default it's fetched from {@link CoderRegistry}, and Beam SQL field types are included,
+   * like Integer/Long/Short/Byte/Float/Double/BigDecimal/Date.
    */
   public Coder<AccumT> getAccumulatorCoder(CoderRegistry registry)
       throws CannotProvideCoderException {
+    //Register coder for Short/Float/BigDecimal/Date
+    registry.registerCoderForClass(Short.class, SerializableCoder.of(Short.class));
+    registry.registerCoderForClass(Float.class, SerializableCoder.of(Float.class));
+    registry.registerCoderForClass(BigDecimal.class, BigDecimalCoder.of());
+    registry.registerCoderForClass(Date.class, SerializableCoder.of(Date.class));
```

Comment:
Adding coders to the registry automatically in a get method feels wrong. The user won't be expecting that, and if for some reason they've registered their own custom coders for those types, they won't be able to use SQL accumulators with them. I'd rather see us look into adding these to the common registry somehow as a separate change, possibly as part of creating a BeamSqlEnv or something similar?